src/convenience/webhook.ts

```diff
@@ -149,14 +97,27 @@ export function webhookCallback<C extends Context = Context>(
             await initCall;
             initialized = true;
         }
-        await timeoutIfNecessary(
-            bot.handleUpdate(await update, webhookReplyEnvelope),
-            typeof onTimeout === "function"
-                ? () => onTimeout(...args)
-                : onTimeout,
-            timeoutMilliseconds,
-        );
-        if (end !== undefined && !usedWebhookReply) end();
+        const { update, respond, end, handlerReturn } = server(...args);
+        let usedWebhookReply = false;
+        const webhookReplyEnvelope: WebhookReplyEnvelope = {
+            send: async (json) => {
+                usedWebhookReply = true;
+                await respond(json);
+            },
+        };
+        try {
+            await timeoutIfNecessary(
+                bot.handleUpdate(await update, webhookReplyEnvelope),
+                typeof onTimeout === "function"
+                    ? () => onTimeout(...args)
+                    : onTimeout,
+                timeoutMilliseconds,
+            );
+            if (end !== undefined && !usedWebhookReply) end();
+        } catch {
+            // Adapter rejected processing an update
+        }
```

Comment:
Also, we should not suppress error, but throw them instead.

Reply:
Does it mean an error occured if the adapter just rejected the request from Telegram (e.g. `requrst.url !== secretPath`)? Or we can create an `class UpdateRejected extends Error` and throw iif `!(error instanceof UpdateRejected)`.

Reply:
We are not going to support secret path matching in this iteration. If an adapter throws, something is wrong.

If we later implement adapters that choose not to process some requests, they must integrate with their respective web frameworks, and e.g. call `next()`.